### PR TITLE
feat(TE-43): Typescript in article puffs with fixed dependencies

### PR DESCRIPTION
**Pull request by bibicollins1**
### Description

This PR includes: 
- Create packages ts-components for new typescript components, ViewCountWrapper has been added to this package
- Create the InArticlePuff component with tests using React testing library

Screenshot 2021-05-04 at 14 02 00
Screenshot 2021-05-04 at 14 02 38
Screenshot 2021-05-04 at 14 03 51
Screenshot 2021-05-04 at 14 03 37

### CODE REVIEW

File: packages/ts-components/src/__tests__/ViewCountWrapper.test.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
+/**
+ * @jest-environment jsdom
```
Comment (vinhlam) [inline packages/ts-components/src/__tests__/ViewCountWrapper.test.tsx:2]: this no longer require as we added it to one of the config files

File: packages/ts-components/src/in-article-puff/styles.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+export const MainContentContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-evenly;
+`;
+
+export const LinkWrapper = styled.a<ContainerType>`
+  display: flex;
+  flex-direction: row;
+  align-items: top;
```
Comment (vinhlam) [inline packages/ts-components/src/in-article-puff/styles.ts:98]: `top` is not a valid value for align-items, its been complaining in my IDE

**Review by vinhlam — APPROVED** [review]
happy for you to not fix these as i can do them in my PR

File: packages/ts-components/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,1565 @@
+# Change Log
```
Comment (flashcheeks) [inline packages/ts-components/CHANGELOG.md:1]: We should delete this file.

File: packages/ts-components/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# @times-components/responsive-image
```
Comment (flashcheeks) [inline packages/ts-components/README.md:1]: We should update this file with relevant info/commands to help explain the components and example usage.

File: packages/ts-components/package.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    "@times-components/storybook": "4.1.60",
+    "@times-components/test-utils": "2.3.8",
+    "@types/jest": "24.9.1",
+    "@types/react-dom": "17.0.3",
+    "@types/react-test-renderer": "16.9.0",
+    "@types/styled-components": "5.1.9",
+    "jest": "24.8.0",
+    "metro-react-native-babel-preset": "0.53.1",
+    "prettier": "1.14.3",
+    "react-native": "0.61.5",
```
Comment (flashcheeks) [inline packages/ts-components/package.json:73]: Do we need native libraries?

File: packages/ts-components/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "outDir": "dist",
+    "rootDir": "src",
+    "jsx": "react-native",
```
Comment (flashcheeks) [inline packages/ts-components/tsconfig.json:6]: Should this be `react`?

File: packages/article-skeleton/src/article-body/article-body.web.js [hunk 7]
```diff
@@ -158,6 +160,17 @@ const renderers = ({
             label={decodeURIComponent(label)}
           />
         );
+      case "in-article-puff":
+        return inArticlePuffFlag ? (
+          <InArticlePuff
```
Comment (flashcheeks) [inline packages/article-skeleton/src/article-body/article-body.web.js:165]: Looking at the current/old `in-article-puff` component I can see that it receives some props but also loads data from an external .json file (receives id then loads `image`, `copy`, `url` etc).

It is probably worth us chatting about our desired data approach.

**Review by flashcheeks — CHANGES_REQUESTED** [review]

File: packages/ts-components/package.json (hunk 5 above)
Comment (bibicollins1) [reply packages/ts-components/package.json:73]: @vinhlam

Comment (vinhlam) [reply packages/ts-components/package.json:73]: Ill have a check to see if it causes any issues

File: packages/ts-components/CHANGELOG.md (hunk 3 above)
Comment (vinhlam) [reply packages/ts-components/CHANGELOG.md:1]: does it not automatically get generated when there are changes and version updates?

File: packages/ts-components/package.json (hunk 5 above)
Comment (vinhlam) [reply packages/ts-components/package.json]: I can remove it from the dependences and the babel config but if I try to do any `yarn add` it complains. I've removed it, and it builds and passes all tests

**Review by flashcheeks — APPROVED** [review]

**Pull request merged**